contracts/Market.sol

```diff
@@ -1776,30 +1510,20 @@ contract Market is
   /**
    * @notice Allocates the removals, amounts, and suppliers needed to fulfill the purchase.
    * @param amount The number of carbon removals to purchase.
-   * @return countOfRemovalsAllocated The number of distinct removal IDs used to fulfill this order.
-   * @return ids An array of the removal IDs being drawn from to fulfill this order.
-   * @return amounts An array of amounts being allocated from each corresponding removal token.
-   * @return suppliers The address of the supplier who owns each corresponding removal token.
+   * @return SupplyAllocationData The removals, amounts, suppliers and count data returned
+   * from the supply allocation algorithm.
    */
   function _allocateSupply(
     uint256 amount
-  )
-    private
-    returns (
-      uint256 countOfRemovalsAllocated,
-      uint256[] memory ids,
-      uint256[] memory amounts,
-      address[] memory suppliers
-    )
-  {
+  ) private returns (SupplyAllocationData memory) {
```

Comment:
Consider making any private functions internal so we can test them in foundry. Rule of thumb applies globally